### ISSUE TITLE
"Optimizing" a class doesn't work right

### ISSUE THREAD

**Issue by NTillmann** [facebook/prepack#1663]
Prepacking 
```js
(function() {
    class foo { }
    __optimize(foo);
    global.foo = foo;
})();
```

Currently results in the following strange message, plus the following prepacked code, which is obviously wrong.
```
Errors found while prepacking
In input file /tmp/test.js(4:18) Warning PP1023: Program may terminate with exception: not callable
  TypeError
    at /tmp/test.js:4:18 (https://github.com/facebook/prepack/wiki/PP1023)
```
```js
(function () {
  var $$0 = {
    enumerable: false,
    configurable: true,
    writable: true
  };

  var _$0 = this;

  var _$1 = _$0.TypeError;
  var _$2 = _$1.prototype;
  var _$3 = _$0.Object;
  var _$4 = _$3.defineProperty;
  var _$5 = _$3.getPrototypeOf;

  var __constructor = function () {};

  var _0 = class {
    constructor() {
      var _5 = _$2;

      var _2 = (__constructor.prototype = _5, new __constructor());

      $$0.value = "TypeError\n    at /tmp/test.js:4:18", _$4(_2, "stack", $$0);
      $$0.value = "not callable", _$4(_2, "message", $$0);

      if (_$5(_2) !== _5) {
        throw new Error("unexpected prototype");
      }

      throw _2;
    }

  };

  foo = _0;
}).call(this);
```

**Comment by cblappert** [comment]
What exactly does optimizing a class mean? Optimizing each of its methods individually?

**Comment by trueadm** [comment]
I was also confused by this, we should probably show a better error, like "classes are not supported as optimized functions". Unless someone actually meant to optimize the constructor – but that doesn't make any sense either.

**Comment by cblappert** [comment]
Resolved by PR #2214
